samba/map.jinja

```diff
@@ -1,47 +1,18 @@
-{% set samba = salt['grains.filter_by']({
-    'Debian': {
-        'server': 'samba',
-        'client': 'samba-client',
-        'service': salt['grains.filter_by']({
-	    'lenny': 'samba',
-	    'squeeze': 'samba',
-	    'wheezy': 'samba',
-	    'jessie': 'smbd',
-	    'default': 'smbd',
-	}, grain='oscodename'),
-        'config': '/etc/samba/smb.conf',
-        'config_src': 'salt://samba/files/smb.conf',
-    },
-    'RedHat': {
-        'server': 'samba',
-        'client': 'samba',
-        'service': 'smb',
-        'config': '/etc/samba/smb.conf',
-        'config_src': 'salt://samba/files/smb.conf',
-    },
-    'CentOS': {
-        'server': 'samba',
-        'client': 'samba',
-        'service': 'smb',
-        'config': '/etc/samba/smb.conf',
-        'config_src': 'salt://samba/files/smb.conf',
-    },
-    'Arch': {
-        'server': 'samba',
-        'client': 'samba',
-        'service': 'smbd',
-        'config': '/etc/samba/smb.conf',
-        'config_src': 'salt://samba/files/smb.conf',
-    },
-    'FreeBSD': {
-        'server': 'samba44',
-        'client': 'samba44',
-        'service': 'samba_server',
-        'config': '/usr/local/etc/smb4.conf',
-        'config_src': 'salt://samba/files/smb.conf',
-    },
-}, merge=salt['grains.filter_by']({
-    'Ubuntu': {
-        'service': 'smbd',
-    },
-}, grain='os', merge=salt['pillar.get']('samba:lookup'))) %}
+{## start with defaults from defaults.yaml ##}
+{% import_yaml "samba/defaults.yaml" as defaults %}
+{% import_yaml "samba/osmap.yaml" as osmap %}
+{% import_yaml "samba/codenamemap.yaml" as oscodenamemap %}
```

Comment:
Initial thoughts: I prefer the single map file over the separate defaults/osmap/codenamemap files. It's nice to glance at a single file to see all available deviations. It makes it straightforward to add a new OS too since you can just copy/paste one of the existing blocks. Plus even in monolithic form, it's just not that long.

Reply:
Do you suggest single map.jinja file be used and the yaml files depreciated.  I have no objections particularly if the abstraction is reusable.

BTW, what is the status of template-formula as I used that as my "template"?

Reply:
A single `map.jinja` is the most canonical for data that varies by OS (but it's not law). A `defaults.yaml` file is a nice pattern for setting defaults for any Pillar-overridable values.